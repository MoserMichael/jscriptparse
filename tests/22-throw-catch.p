# checks formatting of error messages

def rec(num) {
    if num == 0 {
      throw "throwing an error string"
    }
    return rec(num - 1)
}        

try 
    rec(5)
catch ex {
  println("handling exception")
  throw "throw from catch"
}
