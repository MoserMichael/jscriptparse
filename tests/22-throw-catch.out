handling exception
Error: "throw from catch"
#(tests/22-throw-catch.p:14)       throw "throwing an error string"
                           |.......^
#(tests/22-throw-catch.p:13) def rec(num) {
                           |.^
#(tests/22-throw-catch.p:12)     return rec(num - 1)
                           |............^
#(tests/22-throw-catch.p:11) def rec(num) {
                           |.^
#(tests/22-throw-catch.p:10)     return rec(num - 1)
                           |............^
#(tests/22-throw-catch.p:9) def rec(num) {
                          |.^
#(tests/22-throw-catch.p:8)     return rec(num - 1)
                          |............^
#(tests/22-throw-catch.p:7) def rec(num) {
                          |.^
#(tests/22-throw-catch.p:6)     return rec(num - 1)
                          |............^
#(tests/22-throw-catch.p:5) def rec(num) {
                          |.^
#(tests/22-throw-catch.p:4)     return rec(num - 1)
                          |............^
#(tests/22-throw-catch.p:3) def rec(num) {
                          |.^
#(tests/22-throw-catch.p:2)     rec(5)
                          |.....^
#(tests/22-throw-catch.p:1)   throw "throw from catch"
                          |...^
